crates/re_space_view_spatial/src/max_image_dimension_subscriber.rs

```diff
@@ -0,0 +1,96 @@
+use ahash::HashMap;
+use nohash_hasher::IntMap;
+use once_cell::sync::OnceCell;
+use re_data_store::{StoreSubscriber, StoreSubscriberHandle};
+use re_log_types::{EntityPath, StoreId};
+use re_types::{components::TensorData, Loggable};
+
+#[derive(Default, Debug, Clone, PartialEq, Eq, Hash)]
+pub struct ImageDimensions {
+    pub height: u64,
+    pub width: u64,
+    pub channels: u64,
+}
+
+#[derive(Default, Debug, Clone)]
+pub struct MaxImageDimensions(IntMap<EntityPath, ImageDimensions>);
+
+impl MaxImageDimensions {
+    /// Accesses the image dimension information for a given store
+    pub fn access<T>(
+        store_id: &StoreId,
+        f: impl FnOnce(&IntMap<EntityPath, ImageDimensions>) -> T,
+    ) -> Option<T> {
+        re_data_store::DataStore::with_subscriber_once(
+            MaxImageDimensionSubscriber::subscription_handle(),
+            move |subscriber: &MaxImageDimensionSubscriber| {
+                subscriber.max_dimensions.get(store_id).map(|v| &v.0).map(f)
+            },
+        )
+        .flatten()
+    }
+}
```

Comment:
so annoying to ahve to repeat this boilerplate. Store subscriber needs a builtin layer for this...